src/Dialect/ONNX/Transforms/Decompose.cpp

```diff
@@ -694,6 +695,83 @@ bool hasNoActivationConsumer(Value convTransposeResult) {
   return true;
 }
 
+inline bool isTwoFourOrFive(int n) { return n == 2 || n == 4 || n == 5; }
+
+// This decomposition currently do not support all possible convtranspose
+// operations. Below are the supported usecases.
+// 1) stride[2], pads [1,1], kernel 4,  where convtranspose will decompose to
+// two conv operation. with each conv having kernel divided by 2. 2) stride [4],
+// pads [2,2], kernel 8,  where it will decompose into 4 conv operations
+//  each conv with one fourth of the convtranspose kernel size.
+// 3) stride [5], pads [2,2], kernel 10 where it will decompose into 5 conv
+// operations each conv with one fifth of the convtranspose kernel size.
+
+bool ShouldDecomposeConvTransposeOp1dToPhasedConvs(Value convTransposeResult,
+    ArrayAttr kernelShapeAttr, ArrayAttr padsShapeAttr,
+    ArrayAttr stridesShapeAttr, ArrayAttr outputShapeAttr) {
+  ONNXConvTransposeOp op =
+      mlir::cast<ONNXConvTransposeOp>(convTransposeResult.getDefiningOp());
+  bool areSpatialDimsStatic = hasShapeAndRank(convTransposeResult) &&
+                              hasStaticSpatialDims(op.getX()) &&
+                              hasStaticSpatialDims(op.getW());
+  if (!areSpatialDimsStatic)
+    return false;
+  // kernel shape is required for decomposition, Not supporting the case where
+  // kernel shape can be inferred. Not supporting the case where pad values are
+  // to be inferred automatically from outputShape.
+  if (!kernelShapeAttr || outputShapeAttr) {
+    return false;
+  }
+
+  auto kernelShape = getIntVectorFromArrayAttr(kernelShapeAttr);
+  auto padsShape = (padsShapeAttr) ? getIntVectorFromArrayAttr(padsShapeAttr)
+                                   : SmallVector<int64_t>({0, 0});
+  auto stridesShape = (stridesShapeAttr)
+                          ? getIntVectorFromArrayAttr(stridesShapeAttr)
+                          : SmallVector<int64_t>({1});
+
+  RankedTensorType outputType =
+      mlir::cast<RankedTensorType>(convTransposeResult.getType());
+  auto outputShape = outputType.getShape();
+  // Checking to ensure only convtranspose with 1D spatial dims and stride 2, 4
+  // or 5 are supported.
+  if ((outputShape.size() != 3) || (stridesShape.size() != 1) ||
+      (padsShape.size() != 2) || !isTwoFourOrFive(stridesShape[0]))
+    return false;
+  // number of conv phases equals to the stride.
+  int numberOfPhases = stridesShape[0];
+  // If numberOfPhases is not 5, the ofm spatial dim should be evenly divisible
+  // by num of phases.
+  if ((numberOfPhases != 5) &&
+      !(outputShape[outputShape.size() - 1] % numberOfPhases == 0)) {
+    return false;
+  }
+  // The decomposistion creates the ofm of below dim, hence checking if the
+  // original convtranspose ofm dim is matching to the decomposition ofm.
+  // this is specific for convtranspose with stride 5.
+  auto convSpatialDim =
+      std::floor(outputShape[outputShape.size() - 1] / stridesShape[0]);
```

Comment:
I wonder if this is cleaner
```suggestion
      std::floor(outputShape.back() / stridesShape[0]);
```

Reply:
Correct.

Reply:
I will take care of it next time.